### PR TITLE
x/config: More documentation

### PR DESCRIPTION
**Pull request by abhinav**
This adds and updates some of the docs in the x/config package.

Depends on #1020

### CODE REVIEW

File: x/config/doc.go [hunk 1]
```diff
@@ -24,13 +24,20 @@
 //
 // Usage
 //
-// To build a Dispatcher, set up a Configurator and inform it about the
-// different transports that it needs to support. This object is re-usable
-// and may be stored as a singleton in your application.
+// To build a Dispatcher, first create a new Configurator. This object will be
+// responsible for loading your configuration. It does not yet know about the
+// different transports, peer lists, etc. that you want to use. Inform it
+// about them by registering them with it.
```
Comment (akshayjshah) [inline x/config/doc.go]: Last sentence has lots of pronouns. Perhaps drop the last "with it"?

File: x/config/doc.go [hunk 2]
```diff
@@ -179,35 +186,131 @@
...
+// 	    url: https://host/yarpc
+// 	    round-robin:
+// 	      peers:
+// 	        - 127.0.0.1:8080
+// 	        - 127.0.0.1:8081
+// 	        - 127.0.0.1:8082
+//
+// In the example above, the system will round-robin the requests between the
+// different addresses. The specified URL will be used as a template for the
+// HTTP requests made to these hosts.
```
Comment (akshayjshah) [inline x/config/doc.go]: Perhaps clarify that this templating is specific to the HTTP transport, and that other transports may not support the same thing?

File: x/config/chooser.go [hunk 3]
```diff
@@ -29,10 +29,71 @@ import (
...
+// peer list updater (which informs the peer list about different peers),
+// allowing transports to rely on these two pieces for peer selection and load
+// balancing.
+//
+// Format
+//
+// Peer lists may define only one of the following keys: `peer`, `with`, or
+// the name of any registered PeerListSpec.
+//
+// `peer` indicates that requests must be sent to a single peer.
```
Comment (abhinav) [inline x/config/chooser.go]: transport dependent

added comment

File: x/config/chooser.go [hunk 4]
```diff
@@ -29,10 +29,71 @@ import (
...
+// 	  peer: 127.0.0.1:8080
+//
+// `with` specifies that a named peer list preset defined by the transport
+// should be used rather than defining one by-hand in the configuration.
+//
+// 	# Given a dev-proxy preset on your TransportSpec,
+// 	http:
+// 	  with: dev-proxy
+//
+// If the name of a registered PeerListSpec was specified, an object
```
Comment (abhinav) [inline x/config/chooser.go]: changed to,

> If the name of a registered PeerListSpec is the key, an [..]

File: x/config/doc.go (hunk 1 above)
Comment (abhinav) [inline x/config/doc.go]: Change to,

> // You can inform
// the Configurator about the different transports, peer lists, etc. by
// registering them using RegisterTransport, RegisterPeerList, and
// RegisterPeerListUpdater.

**Pull request merged**